samples/matter/lock/src/credentials/credentials_manager_schedules.cpp

```diff
@@ -0,0 +1,372 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "credentials_data_types.h"
+#ifdef CONFIG_LOCK_SCHEDULES
```

Comment:
Wouldn't adding this file to compilation based on this Kconfig make more sense than putting the whole code under ifdef?

Reply:
Because we're using https://github.com/nrfconnect/sdk-nrf/blob/main/samples/matter/lock/CMakeLists.txt#L31 automatic source adding mechanism. @markaj-nordic You wrote this code before, do you think I should split them?

Reply:
I've split it.